interpreter/per-host/src/main/scala/io/buoyant/transformer/perHost/LocalhostTransformer.scala

```diff
@@ -0,0 +1,43 @@
+package io.buoyant.transformer.perHost
+
+import com.twitter.finagle.Name.Bound
+import com.twitter.finagle.{Addr, Address, Name, NameTree}
+import com.twitter.util.Activity
+import io.buoyant.namer.{DelegateTree, DelegatingNameTreeTransformer}
+import java.net.InetAddress
+
+/**
+  * The localhost transformer filters the list of addresses down to only
+  * addresses that have the same IP address as localhost.  This can be used by
+  * an incoming router to only route traffic to local destinations.
+  */
+class LocalhostTransformer extends DelegatingNameTreeTransformer {
+
+  private[this] val localhost = InetAddress.getLocalHost.getAddress
```

Comment:
tioli: per discussion, maybe worth elaboring on the intended use of this class in the docstring.